src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1336,8 +1398,19 @@ ExtractRangeTableId(Node *node)
  * to a form that expandRTE can handle.
  */
 static void
-ExtractColumns(RangeTblEntry *rangeTableEntry, int rangeTableId, List *dependentJobList,
+ExtractColumns(RangeTblEntry *callingRTE, int rangeTableId,
 			   List **columnNames, List **columnVars)
+{
+	expandRTE(callingRTE, rangeTableId, 0, -1, false, columnNames, columnVars);
```

Comment:
For readibility, might be better if we pass those three const args first setting variables for them

Also, I wonder why we pass `include_dropped` to be false

Reply:
I guess we pass that false because we don't want to include the dropped columns in eref-colnames or joinaliasvars. We can also return all vars(including dropped ones) and then filter it while setting eref-colnames and joinaliasvars to simplify the `SetJoinRelatedColumnsCompat`. I can maybe try that as well

Reply:
>I guess we pass that false because we don't want to include the dropped columns in eref-colnames or joinaliasvars

If that is the case, I'm okay with passing it to be false to keep logic simple, but we should double-check the reason behind it

Reply:
https://github.com/citusdata/citus/blob/8efca3b60ac6a28eaf8787a98996ea47b0d5d633/src/backend/distributed/planner/multi_physical_planner.c#L1262-L1272